channel/consensus_channel/consensus_channel.go

```diff
@@ -36,32 +39,39 @@ func newConsensusChannel(
 	myIndex ledgerIndex,
 	outcome LedgerOutcome,
 	signatures [2]state.Signature,
-) (consensusChannel, error) {
+) (ConsensusChannel, error) {
+
+	cId, err := fp.ChannelId()
```

Comment:
Outside the scope of this PR but its annoying that calculating a channelId returns an error.

Reply:
Is there a way around that?

Reply:
We could just `panic` instead of returning an error 😉

Reply:
I actually think returning an error is the right thing to do here, because we need to validate the `FixedPart` and check for an (admittedly almost impossible) hash collision. 

The approach I took here is to do have those checks done when constructing the `consensusChannel`, and then (if successful) cache the channelId. Then, it can be looked up later on (for example inside `MarshalJSON`) without needing to handle an error. 

All of this machinery is there to prevent unexpected `panics`, especially when they could be triggered by the application providing "bad" data.